### ISSUE TITLE
Training and evaluation metrics not being printed in v4.1.0

### ISSUE THREAD

**Issue by martin-wiebusch-thg** [microsoft/LightGBM#6150]
## Description
As of version 4.1.0 the training loss and other evaluation metrics no longer seem to be printed when using the Python 'train' API.

## Reproducible example
```python
import numpy as np
import lightgbm

rng = np.random.default_rng(1234)
n = 1000
x = rng.normal(size=[n, 3])
c = np.array([0.5, -1, 1.5])
p = 1.0/(1.0 + np.exp(-np.dot(x, c)))
y = rng.uniform(size=n) > p
dataset = lightgbm.Dataset(data=x, label=y)

lightgbm.train(
    params={'objective': 'binary'},
    train_set=dataset,
    num_boost_round=10,
    valid_sets=[dataset],
)
```

For lightgbm 3.2.1 this generates the following output:
```
[LightGBM] [Info] Number of positive: 500, number of negative: 500
[LightGBM] [Warning] Auto-choosing col-wise multi-threading, the overhead of testing was 0.002771 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 765
[LightGBM] [Info] Number of data points in the train set: 1000, number of used features: 3
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.500000 -> initscore=0.000000
[1]	training's binary_logloss: 0.649627
[2]	training's binary_logloss: 0.613681
[3]	training's binary_logloss: 0.583392
[4]	training's binary_logloss: 0.557626
[5]	training's binary_logloss: 0.535257
[6]	training's binary_logloss: 0.516249
[7]	training's binary_logloss: 0.500036
[8]	training's binary_logloss: 0.48534
[9]	training's binary_logloss: 0.472088
[10]	training's binary_logloss: 0.459919
```

In lightgbm 4.1.0 I only get
```
[LightGBM] [Info] Number of positive: 500, number of negative: 500
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000103 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 765
[LightGBM] [Info] Number of data points in the train set: 1000, number of used features: 3
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.500000 -> initscore=0.000000
```

## Environment info

LightGBM version or commit hash: 4.1.0

Command(s) you used to install LightGBM

```shell
pip install lightgbm
```

Running on macOS.

## Additional Comments
I have also tried specifying additional validation datasets and metrics. None of the variants I tried printed any metrics.
